server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/DomainQuotaRoutesTest.java

```diff
@@ -447,6 +448,54 @@ void putQuotaShouldUpdateBothQuota() {
         assertThat(maxQuotaManager.getDomainMaxStorage(TROUVÉ_COM)).contains(QuotaSize.size(42));
     }
 
+    @Test
+    void putQuotaWithNegativeCountShouldFail() throws MailboxException {
+        maxQuotaManager.setDomainMaxMessage(TROUVÉ_COM, QuotaCount.count(52));
+        maxQuotaManager.setDomainMaxStorage(TROUVÉ_COM, QuotaSize.size(42));
+        Map<String, Object> errors = given()
+            .body("{\"count\":-5,\"size\":30}")
+            .put(QUOTA_DOMAINS + "/" + TROUVÉ_COM.name())
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400)
+            .contentType(ContentType.JSON)
+        .extract()
+            .body()
+            .jsonPath()
+            .getMap(".");
+
+        assertThat(errors)
+            .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+            .containsEntry("type", "InvalidArgument")
+            .containsEntry("message", "Invalid quota. Need to be an integer value greater or equal to -1");
+
+        assertThat(maxQuotaManager.getDomainMaxMessage(TROUVÉ_COM)).contains(QuotaCount.count(52));
+        assertThat(maxQuotaManager.getDomainMaxStorage(TROUVÉ_COM)).contains(QuotaSize.size(42));
```

Comment:
Asserting the failure when trying to update the domain quota shouldn't be enough? Shouldn't the assertions about the domain quota not being updated in that case moved to an other test? 

Same question for other tests